hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -778,6 +831,38 @@ withRetries :: ShrinkRetries -> Property -> Property
 withRetries n =
   mapConfig $ \config -> config { propertyShrinkRetries = n }
 
+-- | Add a classifier to the test if the predicate is true
+--
+-- @
+--    prop_with_classifier :: Property
+--    prop_with_classifier = property $ do
+--      xs <- forAll $ Gen.list (Range.linear 0 100) Gen.alpha
+--      for_ xs $ \x ->
+--        classify (x == 0) "newborns" $ do
+--        classify (x > 0 && x < 13) "children" $
+--        classify (x > 12 && x < 20) "teens" $
+--          success
+-- @
+classify :: Bool -> String -> PropertyT IO () -> PropertyT IO ()
+classify = cover 0
+
+cover :: Double -> Bool -> String -> PropertyT IO () -> PropertyT IO ()
+cover _ False _ = id
+cover minCoverage True s = PropertyT
+                 . TestT
+                 . mapExceptT (Lazy.mapWriterT addClassification)
+                 . unTest . unPropertyT
```

Comment:
No pattern matching in function definitions please, use an 1if` or a `case` here, also indentation:

```
cover :: Double -> Bool -> String -> PropertyT IO () -> PropertyT IO ()
cover minimumCoverage condition description =
  if condition then
    PropertyT
      . TestT
      . mapExceptT (Lazy.mapWriterT addClassification)
      . unTest
      . unPropertyT
  else
    id
```